pkg/chunk/mem_cache.go

```diff
@@ -31,7 +31,7 @@ type memItem struct {
 }
 
 type memcache struct {
-	sync.Mutex
+	sync.RWMutex
```

Comment:
RWMutex is heavier than Mutex, we should not optimize the two not-important cases, but slow down the important ones (add/load).

Reply:
Got it